GCS_README.rst

```diff
@@ -0,0 +1,31 @@
+PyPI Cloud - GCS Backend
```

Comment:
Thanks for adding docs! It would be better if you put them in the `storage.rst` file, that way it'll show up on pypicloud.rtfd.org. The PR where [CloudFront support was added](https://github.com/mathcamp/pypicloud/pull/71/files) has a good example to go off of.